src/background/providers/hcaptcha.ts

```diff
@@ -24,19 +81,179 @@ export class HcaptchaProvider implements Provider {
         this.callbacks.updateIcon(this.getBadgeText());
     }
 
+    private handleUrl(url: URL) {
+        const reqUrl = url.origin + url.pathname;
+        const isIssuerUrl = !!reqUrl.match(ISSUER_REGEX);
+
+        let isSpendUrl;
+        if (HCAPTCHA_URLS.some((url: string) => reqUrl === url)) {
```

Comment:
to ease reading
```suggestion
        if (HCAPTCHA_URLS.includes(reqUrl)) {
```

you could even use
```suggestion
// test if the URL is not a special hCaptcha url, and if it a valid spend URL.
const isSpendUrl = !HCAPTCHA_URLS.includes(reqUrl) && SPEND_REGEX.test(reqUrl)
```